mw-webapp/src/component/modal/Modal.tsx

```diff
@@ -1,46 +1,40 @@
-import {ReactElement, useState} from "react";
+import {ReactElement} from "react";
 import {Root as DialogRoot} from "@radix-ui/react-dialog";
 import {ModalContent} from "src/component/modal/ModalContent/ModalContent";
-import {ModalTrigger} from "src/component/modal/ModalTrigger/ModalTrigger";
 
 /**
  * Modal props
  */
 interface ModalProps {
 
   /**
-   * The element that triggers the modal.
+   * The content to display within the modal.
    */
-  trigger: ReactElement<HTMLElement>;
+  content: ReactElement<HTMLElement>;
 
   /**
-   * The content to display within the modal.
+   * Controls whether the modal is open or closed.
    */
-  content: ReactElement<HTMLElement>;
+  open: boolean;
 
   /**
-   * Controls whether the model is initially open or closed.
-   * @default false
+   * Function to set the open state.
    */
-  open?: boolean;
+  setOpen: (open: boolean) => void;
 }
 
 /**
  * Modal component
  */
-export const Modal = (props: ModalProps) => {
-  const [open, setOpen] = useState(props.open ?? false);
+export const Modal = ({open, setOpen, content}: ModalProps) => {
```

Comment:
Let's be consistent And avoid destructuring params

Reply:
Done